### ISSUE TITLE
Support for overriding the interactionLayer of an InteractionBehaviour at runtime.

### ISSUE THREAD

**Issue by julienkay** [ultraleap/UnityPlugin#850]
Just wanted to hear your thoughts on whether you think that's something worth supporting.
(For example for procedurally generated objects)

The things I had to change in order to make it work were in InteractionBehaviour.cs in line 709:
`set { _interactionLayer = value; }`
(remove the protected modifier from the property setter)

and added a nullcheck in InteractionManager.cs in line 988:
```
if (_intObjInteractionLayers.ContainsKey(layer)) {
	_intObjInteractionLayers[layer].Remove(intObj);
}
```
(Not sure if that's without side effects though. Might be better to call initLayers() in the property setter above?)

**Comment by nickjbenson** [comment]
I do think this is something worth supporting; I thought it was supported actually, but it might have been functionality we deferred when we first brought in support for arbitrary interaction layers.

If the fix is as easy as this, it'll be easy to add. I'll give it a look for the next milestone and make sure there aren't any side effects we're missing.

**Comment by nickjbenson** [comment]
Following up on this, it looks like one of our in-house apps hit this issue as well, and the fix should just be in #917.

**Comment by nickjbenson** [comment]
Woops! Looks like it wasn't made unprotected after all. Yeesh. Reopened.
